modules/imgproc/src/rotcalipers.cpp

```diff
@@ -179,32 +206,20 @@ static void rotatingCalipers( const Point2f* points, int n, int mode, float* out
     /* all of edges will be checked while rotating calipers by 90 degrees */
     for( k = 0; k < n; k++ )
     {
-        /* sinus of minimal angle */
-        /*float sinus;*/
-
-        /* compute cosine of angle between calipers side and polygon edge */
-        /* dp - dot product */
-        float dp[4] = {
-            +base_a * vect[seq[0]].x + base_b * vect[seq[0]].y,
-            -base_b * vect[seq[1]].x + base_a * vect[seq[1]].y,
-            -base_a * vect[seq[2]].x - base_b * vect[seq[2]].y,
-            +base_b * vect[seq[3]].x - base_a * vect[seq[3]].y,
-        };
-
-        float maxcos = dp[0] * inv_vect_length[seq[0]];
-
         /* number of calipers edges, that has minimal angle with edge */
         int main_element = 0;
 
-        /* choose minimal angle */
-        for ( i = 1; i < 4; ++i )
+        /* choose minimum angle between calipers side and polygon edge */
+        /* calculating the minimum angle using the cosine results in an incorrect angle selection in 18157 test */
```

Comment:
incorrect result is to strong statement. Original solution based on angle comparison produces wrong result due to limited arithmetic accuracy.

Reply:
I propose to replace the comment with more detailed PR description. For example:
"Replaced vector sorting by angle cosine with sorting with custom comparator. The comparator determins vector order by dot project sign to fix floating point accuracy issue with colinear and close to colinear vectors."

Reply:
Fixed the comment in the code. Added your note to PR description (second and third item on the list).